Sources/MountebankSwift/Models/Recreatable.swift

```diff
@@ -0,0 +1,222 @@
+        INDENT += increaseIndent
+        defer {
+            INDENT -= increaseIndent
+        }
+
+        let propertyList = properties
+            .map { Self.propertySwiftString($0, $1) }
+            .filter { !$0.isEmpty }
+
+        switch propertyList.count {
+        case 0:
+            return ".\(enumCaseWithoutAssociatedValues)()"
+        case 1:
+            return ".\(enumCaseWithoutAssociatedValues)(\(propertyList.joined(separator: DELIMITER)))"
+        default:
+            let indent = String(repeating: WHITESPACE, count: INDENT)
+            let string = propertyList.map { indent + $0 }.joined(separator: DELIMITER)
+            return """
+            .\(enumCaseWithoutAssociatedValues)(
+            \(string)
+            \(String(repeating: WHITESPACE, count: INDENT - 1)))
+            """
+        }
+    }
+}
+
+// MARK: - Private Helpers
+
+extension Recreatable {
+    private func structOrClassSwiftString(properties: [(String?, Recreatable?)]) -> String {
+        let increaseIndent = properties.filter { $0.1?.recreatable.isEmpty == false }.count > 1 ? 1 : 0
+        INDENT += increaseIndent
+        defer {
+            INDENT -= increaseIndent
+        }
+
+        let propertyList = properties
+            .map { Self.propertySwiftString($0, $1) }
+            .filter { !$0.isEmpty }
+
+        switch propertyList.count {
+        case 0:
+            return "\(Self.self)()"
+        case 1:
+            return "\(Self.self)(\(propertyList.joined(separator: DELIMITER)))"
+        default:
+            let indent = String(repeating: WHITESPACE, count: INDENT)
+            let string = propertyList.map { indent + $0 }.joined(separator: DELIMITER)
+            return """
+            \(Self.self)(
+            \(string)
+            \(String(repeating: WHITESPACE, count: INDENT - 1)))
+            """
+        }
+    }
+
+    private static func propertySwiftString(_ label: String?, _ value: Recreatable?) -> String {
+        guard let value else {
+            return ""
+        }
+
+        let recreatable = value.recreatable
+        guard !recreatable.isEmpty else {
+            return ""
+        }
+
+        if let label {
+            return "\(label): \(recreatable)"
+        } else {
+            return "\(recreatable)"
+        }
+    }
+}
+
+// MARK: - Extensions on Swift Types
+
+extension String: Recreatable {
+    var recreatable: String {
+        debugDescription
+    }
+}
+
+extension Int: Recreatable {
+    var recreatable: String {
+        description
+    }
+}
+
+extension Double: Recreatable {
+    var recreatable: String {
+        debugDescription
+    }
+}
+
+extension Bool: Recreatable {
+    var recreatable: String {
+        description
+    }
+}
+
+extension Data: Recreatable {
+    var recreatable: String {
+        "Data(base64Encoded: \(base64EncodedString().debugDescription))!"
+    }
+}
```

Comment:
Data is kind of not a nice format to display as a string.. this is the best I could think of.